gradle/dependencies.gradle

```diff
@@ -23,22 +23,26 @@ ext {
 
   // Available if -PscalaVersion is used. This is useful when we want to support a Scala version that has
   // a higher minimum Java requirement than Kafka. This was previously the case for Scala 2.12 and Java 7.
-  availableScalaVersions = [ '2.12', '2.13' ]
+  availableScalaVersions = [ '2.13', '3.0' ]
 }
 
 // Add Scala version
-def defaultScala212Version = '2.12.14'
 def defaultScala213Version = '2.13.6'
+def defaultScala30Version = '3.0.0'
 if (hasProperty('scalaVersion')) {
-  if (scalaVersion == '2.12') {
-    versions["scala"] = defaultScala212Version
-  } else if (scalaVersion == '2.13') {
+  if (scalaVersion == '2.13') {
     versions["scala"] = defaultScala213Version
+  } else if (scalaVersion == '3.0') {
+    versions["scala"] = defaultScala30Version
+    versions["baseScala"] = '3'
+    versions["baseScalaJackson"] = '2.13'
+    versions["postfixScala"] = '3'
+    versions["postfixScalaOptionalVersion"] = '_3'
```

Comment:
@ijuma that is what I meant with not so readable. As both prefixes are only present on Scala 3 and they have different syntax (one with `_`, the other without) I need this somewhat ugly code.

Please note, this might go away eventually if Gradle supports Scala 3: https://github.com/gradle/gradle/issues/16527